### PR TITLE
Allow configuring a specific image for kubectl

### PR DESCRIPTION
**Pull request by dprotaso**
This PR allows helm users to configure a custom `kubectl` image

[x] Installation

### CODE REVIEW

**Comment by dprotaso** [comment]
(fixed commit message typo)

**Review by jwendell — APPROVED** [review]
Changes LGTM, but you need an approval from an actual maintainer.

File: install/kubernetes/helm/istio-init/templates/job-crd-all.yaml
Comment (sdake) [inline install/kubernetes/helm/istio-init/templates/job-crd-all.yaml]: this logic is a little repetitious and difficult to understand (imo).  why not just create a new flag for the kubectl image name?

**Review by sdake — COMMENTED** [review]
think this can be much simpler.  can you try it as a key?

Also, the upstream for installation is https://github.com/istio/installer.  This directory is deprecated as of 1.2, and will be removed in the future.  Please submit your work to installer first and then backport here.

Cheers,
-steve

**Comment by dprotaso** [comment]
> think this can be much simpler. can you try it as a key?

What does this mean exactly?

**Comment by dprotaso** [comment]
> This directory is deprecated as of 1.2, and will be removed in the future. Please submit your work to installer first and then backport here.

Good to know - thanks

**Comment by sdake** [comment]
> > think this can be much simpler. can you try it as a key?
> 
> What does this mean exactly?

like hub and tag, add a kubectl_image or something similar to values.yaml and use that directly defaulting to `kubectl`

File: install/kubernetes/helm/istio-init/templates/job-crd-all.yaml
Comment (morvencao) [reply install/kubernetes/helm/istio-init/templates/job-crd-all.yaml]: +1, this logic is difficult to understand.

**Comment by dprotaso** [comment]
> Please submit your work to installer first and then backport here.

Closing due to ^

**Pull request closed without merging**
